core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,239 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.cli;
+
+import hudson.Extension;
+import hudson.PluginWrapper;
+import hudson.lifecycle.RestartNotSupportedException;
+import jenkins.model.Jenkins;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+import java.io.PrintStream;
+import java.util.List;
+
+/**
+ * Disable one or more installed plugins.
+ * @since TODO
+ */
+@Extension
+public class DisablePluginCommand extends CLICommand {
+
+    @Argument(metaVar = "plugin1 plugin2 plugin3", required = true, usage = "Plugins to be disabled.")
+    private List<String> pluginNames;
+
+    @Option(name = "-restart", aliases = "-r", usage = "Restart Jenkins after disabling plugins.")
+    private boolean restart;
+
+    @Option(name = "-strategy", aliases = "-s", metaVar = "strategy", usage = "How to process the dependant plugins. \n" +
+            "- none: if a mandatory dependant plugin exists and it is enabled, the plugin cannot be disabled (default value).\n" +
+            "- mandatory: all mandatory dependant plugins are also disabled, optional dependant plugins remain enabled.\n" +
+            "- all: all dependant plugins are also disabled, no matter if its dependency is optional or mandatory.")
+    private String strategy = PluginWrapper.PluginDisableStrategy.NONE.toString();
+
+    @Option(name = "-quiet", aliases = "-q", usage = "Be quiet, print only the error messages")
+    private boolean quiet;
+
+    private static final int INDENT_SPACE = 3;
+
+    @Override
+    public String getShortDescription() {
+        return Messages.DisablePluginCommand_ShortDescription();
+    }
+
+    // package-private access to be able to use it in the tests
+    static final int RETURN_CODE_NOT_DISABLED_DEPENDANTS = 16;
+    static final int RETURN_CODE_NO_SUCH_PLUGIN = 17;
+
+    @Override
+    protected void printUsageSummary(PrintStream stderr) {
+        super.printUsageSummary(stderr);
+        stderr.println(Messages.DisablePluginCommand_PrintUsageSummary());
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+
+        // get the strategy as an enum
+        PluginWrapper.PluginDisableStrategy strategyToUse;
+        try {
+            strategyToUse = PluginWrapper.PluginDisableStrategy.valueOf(strategy.toUpperCase());
+        } catch (IllegalArgumentException iae) {
+            throw new IllegalArgumentException(hudson.cli.Messages.DisablePluginCommand_NoSuchStrategy(strategy, String.format("%s, %s, %s", PluginWrapper.PluginDisableStrategy.NONE, PluginWrapper.PluginDisableStrategy.MANDATORY, PluginWrapper.PluginDisableStrategy.ALL)));
+        }
+
+        // disable...
+        List<PluginWrapper.PluginDisableResult> results = jenkins.pluginManager.disablePlugins(strategyToUse, pluginNames);
```

Comment:
Should we check for null or empty lists?

Reply:
You are still not checking if "pluginNames" is null or empty.

Reply:
Added the Nonnull also

Reply:
I am not talking about the annotations, but about checking here if pluginNames == null and then throwing an exception.

Reply:
The parameter is required (`required=true` in the `@Argument` annotation of the class) so it cannot be null. If the user set "" as one of the plugins, the command will result in `NO_SUCH_PLUGIN` for this specific plugin